api/app.py

```diff
@@ -129,6 +129,43 @@ def create_recipe():
 def update_recipe(recipe_id):
     data = request.get_json()
 
+    recipe = db.session.query(Recipe).filter_by(id=recipe_id).first()
+    if not recipe:
+        return jsonify({"error": "Recipe not found"}), 404
+
+    recipe.name = data['name']
+    recipe.instructions = data['instructions']
+    recipe.duration = data['duration']
+
+    pictures = data['pictures']
+    if pictures:
+        recipe.pictures = ','.join(pictures) if isinstance(pictures,list) else pictures
+    if data['instructions']:
+        Ingredient.query.filter_by(recipe_id=recipe_id).delete()
+
+        for ing in data['ingredients']:
+            new_ingredient = Ingredient(
+                name=ing['name'],
+                unit=ing['unit'],
+                quantity=ing['quantity'],
+                recipe_id=recipe.id
+            )
+
+            db.session.add(new_ingredient)
+
+    if data['categories']:
+        recipe.categories.clear()
+        for cat_name in data['categories']:
+            category = db.session.query(Category).filter_by(name=cat_name).first()
+            if not category:
+                category = Category(name=cat_name,color='default')
```

Comment:
Same as for `create_recipe()`, we don't need to create a `category` that doesn't exist here; we'll use the `create_category()` endpoint. Here you can return an error message if the category does not exist, for example:
```python
return jsonify({'error': f'category {cat["name"]} not found'}), 400
```